cmd/maestro/server/healthcheck_server_test.go

```
@@ -0,0 +1,243 @@
+				},
+			}
+
+			server := &HealthCheckServer{
+				instanceDao: instanceDao,
+				instanceID:  "test-instance",
+			}
+			if tt.hasClient {
+				server.sourceClient = &mockCEClient{ready: tt.ceClientReady}
+			}
+
+			req := httptest.NewRequest(http.MethodGet, "/healthcheck", nil)
```

golangci-lint (2.12.2)

[error] net/http/httptest.NewRequest must not be called. use net/http/httptest.NewRequestWithContext

(noctx)